From fca24849fcd4ca0b36f4d5a63e705e9d9e2b01d4 Mon Sep 17 00:00:00 2001
From: Matt Ranostay <matt.ranostay@intel.com>
Date: Fri, 13 Jun 2014 16:45:26 -0700
Subject: [PATCH] xt_qtaguid: fix oops issues on blank interfaces

T100 is passing blank interfaces into iface_stat_create() which is
in turn panicing the system. Return from the function in these cases.

Signed-off-by: Matt Ranostay <matt.ranostay@intel.com>
---
 net/netfilter/xt_qtaguid.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/net/netfilter/xt_qtaguid.c b/net/netfilter/xt_qtaguid.c
index b2f2bf62b2d0..0119b0da93f2 100644
--- a/net/netfilter/xt_qtaguid.c
+++ b/net/netfilter/xt_qtaguid.c
@@ -956,6 +956,9 @@ static void iface_stat_create(struct net_device *net_dev,
 	}
 
 	ifname = net_dev->name;
+	if (!strcmp(ifname, ""))
+		return;
+
 	if (!ifa) {
 		in_dev = in_dev_get(net_dev);
 		if (!in_dev) {
-- 
1.9.1

